system/jlib/jptree.cpp

```diff
@@ -7201,8 +7201,10 @@ class CJSONReaderBase : public CommonReaderBase<X>
         if ('\"'!=nextChar)
             expecting("\"");
         readString(name);
+        //A blank mapping is legal in json, map it to a well defined internal value so external files can be parsed. (E.g. package-lock.json)
         if (!name.length())
-            error("empty JSON id");
+            name.append("__empty__");
```

Comment:
Should be fine.  Note that content with an actual tag named "__empty__" won't work.